mcs/class/System/System.Diagnostics/Process.cs

```diff
@@ -1521,10 +1341,22 @@ public void BeginOutputReadLine ()
 				throw new InvalidOperationException ("Cannot mix asynchronous and synchonous reads.");
 
 			async_mode |= AsyncModes.AsyncOutput;
-			output_canceled = false;
-			if (async_output == null) {
-				async_output = new ProcessAsyncReader (this, stdout_rd, true);
-				async_output.ReadHandler.BeginInvoke (null, async_output);
+
+			if (async_output == null)
+				async_output = new AsyncStreamReader (this, output_stream.BaseStream, new UserCallBack(this.OutputReadNotifyUser), output_stream.CurrentEncoding);
+
+			async_output.BeginReadLine ();
```

Comment:
I don't think this should be allowed to call multiple times (same with BeginReadLine). MSDN says it should throw instead